enumerable.rb

```diff
@@ -0,0 +1,179 @@
+module Enumerable
+  def my_each
+    return to_enum(:my_each) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count])
+      count += 1
+    end
+    arr
+  end
+
+  def my_each_with_index
+    return to_enum(:my_each_with_index) unless block_given?
+
+    arr = self if instance_of?(Array)
+    arr = to_a if instance_of?(Range)
+    arr = flatten if instance_of?(Hash)
+
+    count = 0
+    while count < arr.length
+      yield(arr[count], count)
+      count += 1
+    end
+    arr
+  end
+
+  def my_select
+    return to_enum(:my_select) unless block_given?
+
+    new_arr = []
+    my_each { |it| new_arr << it if yield(it) }
+    new_arr
+  end
+
+  def my_all?(param = nil)
+    if block_given?
+      my_each { |item| return false if yield(item) == false }
+      return true
+    elsif param.nil?
+      my_each { |num| return false if num.nil? || num == false }
+    elsif !param.nil? && (param.is_a? Class)
+      my_each { |num| return false if num.class != param }
+    elsif !param.nil? && param.instance_of?(Regexp)
+      my_each { |_num| return false unless arg.match(n) }
+    else
+      my_each { |_num| return false if n != param }
+    end
+    true
+  end
+
+  def my_any?(param = nil)
+    if block_given?
+      my_each { |item| return true if yield(item) }
+      return true
+    elsif param.nil?
+      my_each { |num| return true if num.nil? || num == true }
+    elsif !param.nil? && (param.is_a? Class)
+      my_each { |num| return true if num.instance_of?(param) }
+    elsif !param.nil? && param.instance_of?(Regexp)
+      my_each { |_num| return true unless arg.match(num) }
+    else
+      my_each { |_num| return true if num != param }
+    end
+    true
+  end
+
+  def my_none?(param = nil)
```

Comment:
enumerables #my_none? when no block or argument is given returns true only if none of the collection members is true
```      Failure/Error: expect(false_array.my_none?).to be true
      
        expected true
             got false
```

enumerables #my_none? when no block or argument is given returns false only if one of the collection members is true
```
      Failure/Error: expect(true_array.my_none?).to be false
      
        expected false
             got true
```

enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class
```
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)
```      

enumerables #my_none? when a class is passed as an argument returns true if none of the collection is a member of such class::Numeric
```
      Failure/Error: expect(array.my_none?(Numeric)).to be array.my_none?(Numeric)
```

enumerables #my_none? when a class is passed as an argument returns false if any of the collection is a member of such class
```
      Failure/Error: expect(array.my_none?(String)).to be array.my_none?(String)
```
  
enumerables #my_none? when a Regex is passed as an argument returns true if none of the collection matches the Regex
```
      Failure/Error: expect(words.my_none?(/z/)).to be words.none?(/z/)
```

enumerables #my_none? when a pattern other than Regex or a Class is given returns true only if none of the collection matches the pattern
```
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
```

enumerables #my_none? when a pattern other than Regex or a Class is given returns false only if one of the collection matches the pattern
```
      Failure/Error: expect(words.my_none?(5)).to be words.none?(5)
```
### `my_none?` should be identical and returns the same thing as ruby's [`none?`](https://ruby-doc.org/core-2.6.4/Enumerable.html#method-i-none-3F). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_none.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))